src/bio/bio_buffer.c

```diff
@@ -637,16 +637,26 @@ rw_completion(void *cb_arg, int err)
 	D_ASSERT(xs_ctxt->bxc_blob_rw > 0);
 	xs_ctxt->bxc_blob_rw--;
 
-	if (biod->bd_result == 0 && err != 0) {
-		biod->bd_result = daos_errno2der(-err);
+	if (biod->bd_update)
+		err = (DAOS_FAIL_CHECK(DAOS_NVME_BIO_WRITE_ERR) ?
+			err : -DER_MISC);
```

Comment:
The outermost brackets are not necessary.

Reply:
Will fix

Reply:
This is not applicable as code has been changed and this checking is done part of error if function below,